src/api/utils/__tests__/apiUtils.test.js

```diff
@@ -0,0 +1,152 @@
+      const localTranslations = {
+        FI: { name: 'foo', address: 'bar' },
+      };
+      const expected = [{ languageCode: 'FI', name: 'foo', address: 'bar' }];
+
+      const result = denormalizeLocalTranslations(localTranslations);
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('mapApiDataToLocalData', () => {
+    it('maps API data to local data format', () => {
+      const apiNode = {
+        id: '1',
+        translations: [
+          {
+            languageCode: 'FI',
+            name: 'foo',
+            address: 'bar',
+            __typename: 'bazType',
+          },
+        ],
+        __typename: 'bazType',
+      };
+      const expected = {
+        id: '1',
+        translations: {
+          FI: { name: 'foo', address: 'bar' },
+        },
+      };
+
+      const result = mapApiDataToLocalData(apiNode);
+      expect(result).toEqual(expected);
+    });
+
+    it('returns null if apiNode is nullish', () => {
+      const result = mapApiDataToLocalData(null);
+      expect(result).toBeNull();
+    });
+  });
+
+  describe('mapLocalDataToApiData', () => {
+    it('maps local data to API data format', () => {
+      const localData = {
+        id: '1',
+        translations: {
+          FI: { name: 'foo', address: 'bar' },
+        },
+      };
+      const expected = {
+        id: '1',
+        translations: [{ languageCode: 'FI', name: 'foo', address: 'bar' }],
+      };
+
+      const result = mapLocalDataToApiData(localData);
+      expect(result).toEqual(expected);
+    });
+  });
+
+  describe('handleApiConnection', () => {
+    it('handles API connection and returns data and total', () => {
+      const connection = {
+        edges: [
+          {
+            node: {
+              id: '1',
+              translations: [
+                { languageCode: 'FI', name: 'foo', address: 'bar' },
+              ],
+            },
+          },
+        ],
+        count: 1,
+      };
+      const expected = {
+        data: [
+          { id: '1', translations: { FI: { name: 'foo', address: 'bar' } } },
+        ],
+        total: 1,
+      };
+
+      const result = handleApiConnection(connection);
+      expect(result).toEqual(expected);
+    });
+
+    it('returns empty data and total 0 if connection is nullish', () => {
+      const result = handleApiConnection(null);
+      expect(result).toEqual({ data: [], total: 0 });
+    });
+  });
+
+  describe('handleApiNode', () => {
+    it('handles API node and returns data', () => {
+      const node = {
+        id: '1',
+        translations: [{ languageCode: 'FI', name: 'foo', address: 'bar' }],
+      };
+      const expected = {
+        data: {
+          id: '1',
+          translations: { FI: { name: 'foo', address: 'bar' } },
+        },
+      };
+
+      const result = handleApiNode(node);
+      expect(result).toEqual(expected);
+    });
+
+    it('returns data with null if node is nullish', () => {
```

Comment:
Same here as in [previous comment](https://github.com/City-of-Helsinki/kukkuu-admin/pull/329/files#r1974929854) i.e. testing only with null → description could mention `null` instead of `nullish`, or this could be tested with other nullish values beside null.